### ISSUE TITLE
gdal_rasterize: Avoid long conversion times on subsets for OGR supported vector formats

### ISSUE THREAD

**Issue by tschmetzer** [qgis/QGIS#47207]
### Feature description

The "Rasterize (Vector to raster)" processing currently does a very expensive intermediate conversion to a shapefile before outputting to the raster file. This costs a lot of time and temporary disk space (on C:\Users<username>\AppData\Local\Temp\processing_XYZaBC).

gdal_rasterize supports subset filtering using the [-where expression] parameter. Columns can be be filtered using the [-sql select_statement] parameter. 

Depending on the amount of data this can save many hours and dozens of GB temporary disk space

This is a feature request as a follow-up of https://github.com/qgis/QGIS/issues/46294

**Comment by gioman** [comment]
@tschmetzer I changed the title because the request is about adding the WHERE parameter, which is unrelated to the internal translation that QGIS does. The translation is to stay (IMO) and is part of the trade off necessary to allow use any vector input. If this impacts very large inputs then user can always use the CLI or the native QGIS tool (not sure there is one, but it should).

**Comment by tschmetzer** [comment]
@gioman Thanks for trying to explain but I am unfortunately not getting the point. What is the advantage of adding the WHERE parameter if the file is converted anyways? What is the use case?
And independently of that why not add an optimisation for OGR supported vector formats? Of course the user could use the CLI in many cases but then what's the point for providing these functions in QGIS? 
If there is a native QGIS native rasterize tool I'd be happy to know :)

**Comment by agiudiceandrea** [comment]
> gdal_rasterize supports subset filtering using the [-where expression] parameter. Columns can be be filtered using the [-sql select_statement] parameter.

In the mean time, you can add those parameters (and others) using the "Additional command-line parameters" parameter of the algorithm.

**Comment by alexbruy** [comment]
> In the mean time, you can add those parameters (and others) using the "Additional command-line parameters" parameter of the algorithm.

I'd say this is a right way of passing any additional parameters, as adding all possible parameters and their combination will result in overloaded UI and lots of additional logic to handle any possible edge cases.

**Comment by gioman** [comment]
> Of course the user could use the CLI in many cases but then what's the point for providing these functions in QGIS?

@tschmetzer there are at least a couple: 1) not all gdal tools functionalities available in QGIS (even less when many years ago we had the "gdal tools" plugin) and 2) allow easily use those tools regardless of the type of the sources. Foe the vast majority of the users that is ok. Again, is a trade off. Advanced users can have a look (if they need) to the command generated in QGIS (and gdal docs) and make the necessary changes to be able tun it from the CLI, with no translations.

**Comment by tschmetzer** [comment]
>> @tschmetzer I changed the title because the request is about adding the WHERE parameter, which is unrelated to the internal translation that QGIS does. The translation is to stay (IMO) and is part of the trade off necessary to allow use any vector input.

> @gioman Thanks for trying to explain but I am unfortunately not getting the point. What is the advantage of adding the WHERE parameter if the file is converted anyways? What is the use case?

@gioman Sorry I'm still not getting that point: Why add a WHERE parameter if the file is to be translated anyways and where/how are you meaning to add it?


>@tschmetzer there are at least a couple: 1) not all gdal tools functionalities available in QGIS (even less when many years ago we had the "gdal tools" plugin) and **2) allow easily use those tools regardless of the type of the sources**. Foe the vast majority of the users that is ok. 

Exactly that's what I would like to achieve with my feature proposal :)

> Again, is a trade off. Advanced users can have a look (if they need) to the command generated in QGIS (and gdal docs) and make the necessary changes to be able tun it from the CLI, with no translations.

Thanks for pointing that out. Why not making the trade-off redundant optimizing the functionality provided such that every user can have an easy access to it? From a usability perspective that would make things way easier and save a lot of time for people who aren't experts. I don't see it is that complicated and I would try to contribute (implement) that part.


>>In the mean time, you can add those parameters (and others) using the "Additional command-line parameters" parameter of the algorithm.

>I'd say this is a right way of passing any additional parameters, as adding all possible parameters and their combination will result in overloaded UI and lots of additional logic to handle any possible edge cases.

@alexbruy Of course there is always a general risk with edge cases when implementing new features. Although the logic I am thinking of seems pretty straight forward to me what do people thing about marking a functionality as _experimental_ until it's been tested thoroughly and proven stable?

### To recap and summarize my idea
Add an option to the dialog e.g.:
- [x] Pass filter and selected columns directly to external provider (experimental)

This would apply to gdal_rasterize. With that option set for filters or/and prevously selected columns there would be no conversion. Filters are automatically passed to the -WHERE argument and previously selected colums to the -SQL select_statement.
For individually selected features a conversion will be done.

**Comment by gioman** [comment]
> Why add a WHERE parameter if the file is to be translated anyways and where/how are you meaning to add it?

@tschmetzer no, if there is no QGIS filter in place the input is not translated. But adding the where option would allow you to filter the input in a different way, a way that will not trigger the translation. And one could argue that is possible to add the "where" in the "additional command line parameters" and that this request make no sense then.

> Although the logic I am thinking of seems pretty straight forward to me what do people thing about marking a functionality as _experimental_ until it's been tested thoroughly and proven stable?

you can create an experimental plugin first.
